core/src/impl/KokkosExp_Host_IterateNestLoopWoTile.hpp

```diff
@@ -0,0 +1,626 @@
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright Contributors to the Kokkos project
+
+#ifndef KOKKOS_HOST_EXP_ITERATE_NESTLOOPWOTILE_HPP
+#define KOKKOS_HOST_EXP_ITERATE_NESTLOOPWOTILE_HPP
+
+#include <type_traits>
+
+#include <Kokkos_Layout.hpp>
+#include <Kokkos_Macros.hpp>
+
+namespace Kokkos {
+namespace Impl {
+
+// MDRangePolicy iteration via a nested loop without tiles
+
+// ParallelFor macros: Non-Tagged
+
+#define KOKKOS_IMPL_NLWOTILE_APPLY(func, ...) func(__VA_ARGS__);
+
+// Non-Tagged: Outer Iterate::Right
+#define KOKKOS_IMPL_NLWOTILE_R1(type, rp, func, rank, ...)   \
+  for (auto i0 = static_cast<type>(rp.m_lower[rank - 1]);    \
```

Comment:
```suggestion
  KOKKOS_ENABLE_IVDEP_MDRANGE
  for (auto i0 = static_cast<type>(rp.m_lower[rank - 1]);    \
```

Probably, we need to encourage vectorization

Reply:
Yes, I will be trying this next.